client/llb/state.go

```diff
@@ -236,6 +236,9 @@ func (s State) WithImageConfig(c []byte) (State, error) {
 			WorkingDir string   `json:"WorkingDir,omitempty"`
 			User       string   `json:"User,omitempty"`
 		} `json:"config,omitempty"`
+		Architecture string `json:"architecture"`
+		Variant      string `json:"variant,omitempty"`
+		OS           string `json:"os"`
 	}
```

Comment:
Why not just use https://github.com/opencontainers/image-spec/blob/02efb9a75ee11e05937b535cc5f228f9343ab2f5/specs-go/v1/config.go#L82 ?

Overhead seems negligible